src/portfolios/portfolios.controller.ts

```diff
@@ -211,4 +215,46 @@ export class PortfoliosController {
 
     return createPortfolioModel(portfolio, did);
   }
+
+  @ApiOperation({
+    summary: 'Get Portfolio creation event data',
+    description:
+      'The endpoint retrieves the identifier data (block number, date and event index) of the event that was emitted when the given Numbered Portfolio was created. This requires Polymesh GraphQL Middleware Service',
+  })
+  @ApiParam({
+    name: 'did',
+    description: 'The DID of the Identity whose Portfolio creation event is to be fetched',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiParam({
+    name: 'id',
+    description:
+      'The ID of the portfolio for which Portfolio creation event is to be fetched. Throws an error if default Portfolio (0) details are requested',
+    type: 'string',
+    example: '1',
+  })
+  @ApiOkResponse({
+    description: 'Details of event where the Numbered Portfolio was created',
+    type: SubsidyModel,
+  })
+  @ApiBadRequestResponse({
+    description: 'Event details for default Portfolio is requested',
+  })
+  @ApiNoContentResponse({
+    description: 'Data is not ready by the time it is requested',
+  })
+  @ApiNotFoundResponse({
+    description: "The Portfolio doesn't exist",
+  })
+  @Get('/identities/:did/portfolios/:id/created-at')
+  async createdAt(@Param() { did, id }: PortfolioDto, @Res() res: Response): Promise<void> {
+    const result = await this.portfoliosService.createdAt(did, id);
+
+    if (result) {
+      res.status(HttpStatus.OK).json(new EventIdentifierModel(result));
+    } else {
+      res.status(HttpStatus.NO_CONTENT).send({});
```

Comment:
Throwing an error code seems unfair here as there isn't a problem with the request but the data is not present at the moment. NO_CONTENT was more fitting IMO

Reply:
If you still think so @polymath-eric that this should be NOT_FOUND, I can make the change

Reply:
I can't think of a situation where a user would want to forward the no data onto a service.

I think Not Found makes it easier and more consistent. If the data isn't available then the service should respond as such.

Returning a no content here is going to be annoying to integrate against. I have only seen No Content be used to accept requests and never for a GET.

Reply:
Consider the common way of writting a "GET" helper:

```js

  const response = fetch(url)
  if(response.status >= 300) {
     throw new Error('non 200 response') 
  }
  return response.data // expected to be defined
}
```

If there were a no-content returned for a POST response  then it would take even more code to distinguish when or when not no-content is OK.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204